test/PaymasterCommitment.test.ts

```diff
@@ -189,6 +189,53 @@ contract('Paymaster Commitment', function ([_, relayOwner, relayManager, relayWo
       assert.closeTo(paid, gasUsed, 50)
     })
 
+    it('paymaster should not pay for requests exceeding msg.data size limit', async () => {
+      const r = await makeRequest(web3, {
+        request: {
+          // nonce: '4',
+          data: recipientContract.contract.methods.emitMessage('').encodeABI()
+        },
+        relayData: { paymaster }
+
+      }, sharedRelayRequestData, chainId, forwarderInstance)
+      const gasAndDataLimits = await paymasterContract.getGasAndDataLimits()
+      const hugeApprovalData = '0x' + 'ef'.repeat(parseInt(gasAndDataLimits.calldataSizeLimit))
+      await expectRevert(
+        relayHubInstance.relayCall(10e6, r.req, r.sig, hugeApprovalData, externalGasLimit, {
+          from: relayWorker,
+          gas: externalGasLimit,
+          gasPrice
+        }), 'msg.data exceeded limit'
+      )
+    })
+
+    it('paymaster should not pay for requests with max msg.data size if it rejects in pre', async () => {
+      await paymasterContract.setRevertPreRelayCall(true)
```

Comment:
this is not the case Oleksii raised: preRelayedCall should succeed, and then forwarder should fail (on nonce error)
in this case, 22000 bytes of approvalData makes the paymaster pay (=relayCall emits "TransactionRelayed" instead of "RejectedByPaymaster"

Reply:
See that below